### ISSUE TITLE
error: Failed to link vite: EBUSY

### ISSUE THREAD

**Issue by zhihengGet** [oven-sh/bun#12440]
### What version of Bun is running?

 v1.1.18 (5a0b9352)

### What platform is your computer?

windows 11

### What steps can reproduce the bug?

when vite app is running then do bun update in another cmd , i got Failed to link vite: busy, i should be able to update/add packages during vite dev 

### What is the expected behavior?

 should be able to update/add packages during vite dev 

1. bun dev in my svelte project
2. bun update/ bun install

**Comment by cliffordkleinsr** [comment]
Also getting this issue on windows 11. Only remedy is restarting the IDE

**Comment by Rope-a-dope** [comment]
Save problem here. Window 10 Pro bun v1.1.18

**Comment by ethanforvest** [comment]
> Also getting this issue on windows 11. Only remedy is restarting the IDE

Mine was to stop the dev server

**Comment by Huliiiiii** [comment]
```
> bun update yarn -g
error: Failed to link bun: EBUSY
> bun
bun: The term 'bun' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```
Bun kill itself.

**Comment by Franklivania** [comment]
> ### What version of Bun is running?
> 
> v1.1.18 ([5a0b935](https://github.com/oven-sh/bun/commit/5a0b93523134ca0043ee7ae84f01003aa4b7ff69))
> ### What platform is your computer?
> 
> windows 11
> ### What steps can reproduce the bug?
> 
> when vite app is running then do bun update in another cmd , i got Failed to link vite: busy, i should be able to update/add packages during vite dev
> ### What is the expected behavior?
> 
> should be able to update/add packages during vite dev
> 
>     1. bun dev in my svelte project
> 
>     2. bun update/ bun install
> 
> 
> ### What do you see instead?
> 
> _No response_
> ### Additional information
> 
> _No response_

So, this is what I have figured out... I am using Vite + React + Typescript. I had my server running on one bash terminal, and I opened another to try to install a package. It showed the same error as yours. But if you kill your server _as I did mine_, you would be able to install your dependencies, or do whatever you want to do.

_PS: I know this is not a probable solution, but I presume it can be managed till they sort themselves out there_
Cheers, and happy coding 🍻

**Comment by Atulin** [comment]
Same with Nuxt 3, killing the server also allows for installing the package.

**Comment by crazytweeks** [comment]
Same ussue with me. Am running next dev in another terminal.

But even when nextjs is not running, i will get 'Failed to get typescript: EBUSY' error.

Am using win11
Bun Version :  1.1.24

``
In pnpm i can install any packages those are necessory for my project while dev server is running. its pretty much needed functionality. I dont want to kill running dev server and close vscode and run install command everytime i need a package for my development. 
``

**Comment by sayrusgit** [comment]
The same one with my project when I use nestjs. It just rejects to work when any other bun process is working (e. g. nestjs dev server, vite + react, etc), but works fine if I shutdown all current processes.

![image](https://github.com/user-attachments/assets/7ffc75c0-6926-4dbd-beee-269d7f06afd3)

Honestly this issue has to be fixed ASAP, 'cause it slows down development pretty much.

**Comment by qsaheeb** [comment]
A workaround solution works for me by stopping localhost server first then installing the package.

**Comment by avnshsingh** [comment]
Workaround is working thanks @qsaheeb . But it should be fixed asap

**Comment by Fesyse** [comment]
Same issue right there
```bash
➜ bun ui add form
$ bunx shadcn-ui@latest add form
⠙ Installing form...Command failed with exit code 1: bun add @radix-ui/react-label @radix-ui/react-slot @hookform/resolvers zod react-hook-form
[0.56ms] ".env"
bun add v1.1.24 (85a32991)
Resolving dependencies
Resolved, downloaded and extracted [17]
error: Failed to link next: EBUSY
error: script "ui" exited with code 1
```
in the second terminal im running nextjs server
Shutting down server fixes problem, but that's not a good solution

**Comment by IsaacAndra** [comment]
I noticed that when bun is already running and you try to install some package, this error appears. In my case, bun was running my app locally. So I just stopped it from running and tried to install a package again. Wich btw, worked.

**Comment by JohnC0de** [comment]
For anyone comming from web search (like me):

Dylan fixed it on #13563 and will be included in bun v1.1.27
You can try the fix now with canary (bun upgrade --canary)

Now on build you can run something with bun and also download deps, example
Terminal 1: bun dev
Terminal 2: bunx shadcn init

**Comment by CodeBreaker518** [comment]
tysm @JohnC0de!
